--- cmake/nrf5.cmake	2021-01-02 23:27:03.837000000 +0000
+++ cmake/nrf5_new.cmake	2021-01-02 23:26:05.637152544 +0000
@@ -24,8 +24,8 @@
 include("nrf5_helpers")
 
 # Check nRF SDK
-set(NRF5_SDK_PATH "" CACHE PATH "Path to the nRF5 SDK")
-set(NRF5_SDK_VERSION "" CACHE STRING "nRF5 SDK version, e.g.: 15.3.0")
+set(NRF5_SDK_PATH "REPLACE_SDK_IN_SOURCE_DIR")
+set(NRF5_SDK_VERSION "16.0.0")
 
 if(NOT NRF5_SDK_PATH)
   message(FATAL_ERROR "Path to the nRF5 SDK not provided. Please set NRF5_SDK_PATH cache variable.")
@@ -68,7 +68,7 @@
   add_compile_options(${local_target_flags})
   add_link_options(${local_target_flags})
   if(NRF5_BOARD_DEFINE)
-    add_compile_options("-D${NRF5_BOARD_DEFINE}")
+    add_definitions("-D${NRF5_BOARD_DEFINE}")
   endif()
   nrf5_split_target(${NRF5_TARGET} NRF5_TARGET_FAMILY NRF5_TARGET_VARIANT NRF5_TARGET_GROUP)
 else()
@@ -76,6 +76,8 @@
 endif()
 
 message(STATUS "Using nRF5 target: ${NRF5_TARGET}")
+string(TOUPPER ${NRF5_TARGET} NRF5_TARGET_DEFINE)
+add_definitions("-D${NRF5_TARGET_DEFINE}")
 
 # Check SoftDevice variant.
 set(NRF5_SOFTDEVICE_VARIANT "" CACHE STRING "SoftDevice variant. Set to 'blank' if SoftDevice is not used, otherwise specify version, e.g.: 's130'")
@@ -157,7 +159,7 @@
   if(NOT NRF5_STACK_SIZE MATCHES "^[1-9][0-9]*$")
     message(FATAL_ERROR "NRF5_STACK_SIZE must be a positive decimal integer")
   endif()
-  add_compile_options("-D__STACK_SIZE=${NRF5_STACK_SIZE}")
+  add_definitions("-D__STACK_SIZE=${NRF5_STACK_SIZE}")
   message(STATUS "Using stack size: ${NRF5_STACK_SIZE}")
 endif()
 
@@ -167,7 +169,7 @@
   if(NOT NRF5_HEAP_SIZE MATCHES "^[1-9][0-9]*$")
     message(FATAL_ERROR "NRF5_HEAP_SIZE must be a positive decimal integer")
   endif()
-  add_compile_options("-D__HEAP_SIZE=${NRF5_HEAP_SIZE}")
+  add_definitions("-D__HEAP_SIZE=${NRF5_HEAP_SIZE}")
   message(STATUS "Using heap size: ${NRF5_HEAP_SIZE}")
 endif()
 
@@ -282,40 +284,10 @@
   )
   # Print size information after build
   add_custom_command(TARGET ${exec_target} POST_BUILD
-    COMMAND ${CMAKE_SIZE_BIN} "${exec_target}"
-  )
-  # Targets for creating Intel HEX and binary executable forms
-  add_custom_target(hex DEPENDS ${exec_target} COMMAND ${CMAKE_OBJCOPY_BIN} -O ihex "${exec_target}" "${exec_target}.hex")
-  add_custom_target(bin DEPENDS ${exec_target} COMMAND ${CMAKE_OBJCOPY_BIN} -O binary "${exec_target}" "${exec_target}.bin")
-  # Target for flashing SoftDevice
-  if(local_sd_hex_file_path)
-    add_custom_target(flash_softdevice
-      COMMAND ${NRF5_NRFJPROG} --program ${local_sd_hex_file_path} -f nrf52 --sectorerase ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-      COMMAND ${NRF5_NRFJPROG} --reset -f nrf52 ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-    )
-  else()
-    add_custom_target(flash_softdevice COMMAND true COMMENT "Using a non-SoftDevice configuration (${NRF5_SOFTDEVICE_VARIANT}). Nothing to flash.")
-  endif()
-  # Target for flashing MBR
-  if(local_mbr_hex_file_path)
-    add_custom_target(flash_mbr
-      COMMAND ${NRF5_NRFJPROG} --program ${local_mbr_hex_file_path} -f nrf52 --sectorerase ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-      COMMAND ${NRF5_NRFJPROG} --reset -f nrf52 ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-    )
-  else()
-    if(local_sd_hex_file_path)
-      add_custom_target(flash_mbr COMMAND true COMMENT "Using a SoftDevice configuration (${NRF5_SOFTDEVICE_VARIANT}). Build \'flash_softdevice\' target to flash both MBR and SoftDevice.")
-    else()
-      add_custom_target(flash_mbr COMMAND true COMMENT "Using a non-SoftDevice configuration (${NRF5_SOFTDEVICE_VARIANT}). Nothing to flash.")
-    endif()
-  endif()
-  # Target for flashing the output executable
-  add_custom_target(flash DEPENDS hex 
-    COMMAND ${NRF5_NRFJPROG} --program "${exec_target}.hex" -f nrf52 --sectorerase ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-    COMMAND ${NRF5_NRFJPROG} --reset -f nrf52 ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
-  )
-  # Target for erasing Flash memory and the UICR page
-  add_custom_target(erase_all
-    COMMAND ${NRF5_NRFJPROG} --eraseall -f nrf52 ${nrfjprog_jlink_sn_opt} ${nrfjprog_jlink_sn_arg}
+    COMMAND $ENV{SIZE} "${exec_target}"
+    COMMAND $ENV{OBJCOPY} -O ihex "${exec_target}" "${exec_target}.hex"
+    COMMAND $ENV{OBJCOPY} -O binary "${exec_target}" "${exec_target}.bin"
+    WORKING_DIRECTORY ${OUTPUT_BIN_DIR}
   )
+
 endfunction()
